src/assignments/ray/lesson_4_answers.clj

```diff
@@ -0,0 +1,61 @@
+(ns lesson-4-answers)
+
+;;;; Chapter 9 Answers
+
+
+;;;; Chapter 10 Answers
+
+;; Question 1
+
+(def a-number (atom 0))
+
+(swap! a-number inc)
+(swap! a-number inc)
+(swap! a-number inc)
+
+@a-number
+
+;; Question 2
+
+
+;; Question 3
+
+;; Defining a warrior and priest structure, each character has a map of hit-point and inventory
+;; Hit-point represents another map which contains current hp and total hp, and the amount of hp as value
+;; Inventory repressents a map with what items are inside inventory and value represnts the amount of the item
+(def warrior (ref {:hit-point {:current 15 :total 40} :inventory {:healing-potion 0}}))
+(def priest (ref {:hit-point {:current 30 :total 30} :inventory {:healing-potion 1}}))
```

Comment:
Not sure if it is a good idea to explain too explicitly, feels like a reader can clearly see it from the code?